core/src/ipld_ext.rs

```diff
@@ -0,0 +1,64 @@
+use anyhow::Result;
+use cid::Cid;
+use serde::Serialize;
+
+/// IpldExt is a trait for serialization, deserialization, CID calculation assuming dagcbor.
+pub trait IpldExt: Serialize {
+    /// serde serialize to dag-json String.
+    fn to_json(&self) -> Result<String> {
```

Comment:
nit, these methods should return `Result<Self::Ok, Self::Error>` since we are explicitly extending the serde::Serialize trait

Reply:
https://docs.rs/serde/latest/serde/trait.Serialize.html